blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClientBuilder.scala

```diff
@@ -57,7 +58,8 @@ sealed abstract class BlazeClientBuilder[F[_]] private (
     val executionContext: ExecutionContext,
     val scheduler: Resource[F, TickWheelExecutor],
     val asynchronousChannelGroup: Option[AsynchronousChannelGroup],
-    val channelOptions: ChannelOptions
+    val channelOptions: ChannelOptions,
+    val customDnsResolver: Option[RequestKey => Either[Throwable, InetSocketAddress]]
```

Comment:
Something for 1.0: we'll probably want to refine this to the ip4s address type once it becomes public API.

Well, maybe not, if blaze is just using `java.net` internally.  It's a bigger issue for things with deeper fs2 integration.  Anyway, something to think about.